lib/vdsm/storage/lvm.py

```diff
@@ -782,14 +781,13 @@ def _updatelvs_locked(self, lvs_output, vg_name):
 
         return updatedLVs
 
-    def _update_stale_lvs_locked(self, vg_name):
+    def _update_stale_lvs_locked(self, vg_name, lv_names):
         """
         Check if any LV is stale after an LVM command and make it Unreadable.
         Log a warning if any LV has been made Unreadable.
         Must be called while holding the lock.
         """
         updatedLVs = {}
-        lv_names = (lvn for vgn, lvn in self._lvs if vgn == vg_name)
```

Comment:
With the rebase this commit makes little sense now. I might just change this line to a list to be consistent with the other updates that manage a list, or remove the commit entirely.

Reply:
Lets remove the commit.

Yes, creating the list once is nice, but the code is nicer when we don't have to do
this, and we should really fix the cache structure so creating the list of lvs is
not complicated and expensive as it is now. Once we do this we don't need to be
clever about this.

For example if we had:

     cache = {
         "vgs": [
             "vg-name-1": {
                 "lvs": {
                     "lv-name-1": ...
                     ...
                 }
             ...             
             }
              ...
         ]
         "pvs": {
             "path": ...,
             ...
         }

Getting vg lvs would be:

    cache["vg-name-1"["lvs"]

Getting all lvs is chaining the lvs from all vgs.

The current cache try to make it easy to have flag list of lvs, we don't need to make it easy since we should never do operations on all lvs, only on certain vg.

Reply:
The lv names are accessible via the vg, not the lv objects. We can change VG to be
a class with mutable lvs list and store there the LV named tuple, or have something like:

    class VGInfo:
       def __init__(self, vg, lvs):
           self.vg = vg
           self.lvs = lvs

And store the VGInfo object in the cache instead of the VG objects.

This makes it trivial to invalidate a vg and all the lvs from the cache,
or access one VG lvs.

Reply:
True, contains LV names not the objects. Still, it could be used to get the LVs in a VG without having to loop through them all. But I agree, it is better to change the design to some other option before workaround the current design.

I think I'll write an issue to keep it in mind.

Reply:
Here it goes! https://github.com/oVirt/vdsm/issues/212